### PR TITLE
Improvement to ck integration

### PR DESCRIPTION
**Pull request by pfultz2**
- Add a CI job to test CK
- Add `MIGRAPHX_TUNE_CK` env variable to only do tuning for CK
- Continue tuning even when there is invalid configs
- Fix a bug with parallel compilation not using all available threads
- Add additional test for gemms using half types
- Removed int32 as supported type since it doesnt pass our test suite

### CODE REVIEW

File: src/targets/gpu/compile_ops.cpp [hunk 1]
```diff
@@ -185,7 +203,10 @@ void par_compile(std::size_t n, F f)
 {
     if(n == 0)
         return;
-    par_for(n, n / value_of(MIGRAPHX_GPU_COMPILE_PARALLEL{}, n), f);
+    auto d = value_of(MIGRAPHX_GPU_COMPILE_PARALLEL{});
```
Comment (umangyadav) [inline src/targets/gpu/compile_ops.cpp:206]:
```suggestion
    auto d = value_of(MIGRAPHX_GPU_COMPILE_PARALLEL{}, n);
```

File: src/targets/gpu/jit/ck_gemm.cpp [hunk 2]
```diff
@@ -436,8 +435,10 @@ struct ck_gemm_compiler : compiler<ck_gemm_compiler>
     }
 
     optional<tuning_config>
-    get_tuning_config(context& ctx, instruction_ref ins, const operation& op) const
+    get_tuning_config(context& ctx, instruction_ref ins, const operation& op, bool exhaustive) const
     {
+        if(not exhaustive and not enabled(MIGRAPHX_TUNE_CK{}))
```
Comment (umangyadav) [inline src/targets/gpu/jit/ck_gemm.cpp:440]:
```suggestion
        if(not exhaustive or not enabled(MIGRAPHX_TUNE_CK{}))
```

File: src/targets/gpu/compile_ops.cpp (hunk 1 above)
Comment (pfultz2) [reply src/targets/gpu/compile_ops.cpp:206]: That wont work, because the `n` will be cached when `MIGRAPHX_GPU_COMPILE_PARALLEL` is not set.

**Comment by pfultz2** [comment]
@turneram Any feedback?

**Review by umangyadav — APPROVED** [review]

**Review by turneram — APPROVED** [review]
If we want to leave int32 enabled, then adding the requirement that m, n, and k are all divisible by 4 appears to maintain correctness. We could also investigate further to get more precise criteria and then add it back in another PR.

**Pull request merged**